scanner/src/main/kotlin/experimental/ProvenanceTree.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.experimental
+
+import org.ossreviewtoolkit.model.KnownProvenance
+import org.ossreviewtoolkit.model.RepositoryProvenance
+
+/**
+ * This class contains information about a [root] provenance and all nested [subRepositories].
+ */
+data class ProvenanceTree(
+    /**
+     * The root provenance.
```

Comment:
What *is* a "root provenance" (I assume it's the super-project), and why is it just `KnownProvenance` instead of `RepositoryProvenance` (as there probably cannot be a thing like a `ArtifactProvenance` that refers to multiple child `RepositoryProvenance`)?

Reply:
It's the provenance that contains the nested provenances, I have added this to the comment. The reason that root is a `KnownProvenance` is because this makes the code using a lot simpler because it only has to deal with a single class. It's documented in the `subRepositories` property that it is always empty if `root` is an `ArtifactProvenance`.